snippet_ranger/model2/base_decompose.py

```diff
@@ -0,0 +1,69 @@
+from typing import Iterable
+
+from modelforge import Model
+from ast2vec import Model2Base
+
+
+class Model2BaseDecompose(Model2Base):
+    """
+    Class for model entry filtration, decomposition, filtration again and merging.
+    Produce exactly one model from one, but it has more entries because of decomposition.
+    MODEL_FROM should be iterable.
+    """
+    def __init__(self, *args, **kwargs):
```

Comment:
I am not asking what you think, I am asking why :) The general rule of thumb: KISS